### PR TITLE
New references - Update offer confirmation page (with feature flag)

### PR DESCRIPTION
**Pull request by JR-G**
## Context

This PR was originally raised in [#7359](https://github.com/DFE-Digital/apply-for-teacher-training/pull/7359). However, this PR was reverted due to the fact this content was not behind a feature flag.

### CODE REVIEW

File: app/components/candidate_interface/application_dashboard_course_choices_component.rb [hunk 1]
```diff
@@ -105,6 +105,7 @@ def conditions_row(application_choice)
           OfferConditionsReviewComponent.new(
             conditions: application_choice.offer.conditions_text,
             provider: application_choice.current_course.provider.name,
+            application_form: application_choice.application_form,
```
Comment (JR-G) [inline app/components/candidate_interface/application_dashboard_course_choices_component.rb:108]: I've done this in order to get the feature flag method on the conditions component. Not a huge fan of this... it is temporary but interested to hear alternatives?

**Review by tomas-stefano — APPROVED** [review]
LGTM 👍

File: app/components/candidate_interface/application_dashboard_course_choices_component.rb (hunk 1 above)
Comment (tomas-stefano) [reply app/components/candidate_interface/application_dashboard_course_choices_component.rb:108]: Not a huge fan either, but unfortunately, I think is necessary to be there.

**Comment by JR-G** [comment]
[References: Update offer confirmation page](https://trello.com/c/YEhUDw1A/434-references-update-offer-confirmation-page)

**Pull request merged**